### ISSUE TITLE
iOS: missing dSYM for TitaniumKit framework

### ISSUE THREAD

**Issue by m1ga** [tidev/titanium-sdk#14157]
### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

Report from Slack (but I saw this one too):

When uploading an app to the Apple Store it will show this warning

> The archive did not include a dSYM for the TitaniumKit.framework with the UUIDs [xxxxxx-xxxxxx-xxxx-xxxx-xxxxxxxxxxxx]. Ensure that the archive's dSYM folder includes a DWARF file for TitaniumKit.framework with the expected UUIDs.

You can still upload the app and it will be released

### Expected Behavior

Not seeing the warning

### Actual behavior

Warning is shown

### Reproducible sample

Build an app and upload it to the store

### Steps to reproduce

Build app
upload it to the Apple Store

### Platform

iOS

### SDK version you are using

12.5.1.GA

**Comment by kerrydh** [comment]
Seeing the same in my environment.

**Comment by m1ga** [comment]
Possible solution: https://github.com/AppLovin/AppLovin-MAX-SDK-iOS/issues/387#issuecomment-2746767056
maybe that can be automated in the build process? 

It's still only a warning and you can upload the apps without issues. But it would be awesome not to have the warning and scare people 😄

**Comment by hbugdoll** [comment]
I guess this warning only appears when _"Upload your app's symbols to receive symbolicated reports from Apple"_ is selected during distribution process in Xcode Organizer, because I have never seen this message before 😄 

**Edit:** Yes, that's the reason.

**Comment by hbugdoll** [comment]
I tried `dsymutil` on several `TitaniumKit` binaries (in `iphone/TitaniumKit/build/...` and in `~/Library/Developer/Xcode/DerivedData/...`) built with build setting `DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym"`:

```
warning: no debug symbols in executable
```

~~The challenge would be to add proper debug symbols...~~
https://developer.apple.com/documentation/xcode/building-your-app-to-include-debugging-information

**Edit:** I used `dsymutil` in wrong manner. `dSYMs` are already created for the 3 archives.
Maybe we just need to add `-debug-symbols` flags to the `xcodebuild -create-xcframework` command...

**Comment by m1ga** [comment]
@hansemannn just saw your commit: https://github.com/tidev/titanium-sdk/commit/49531568b4226eed0e5eb84893f7a0f4bfa4da87 
Is that fixing this issue here? If so it can be closed

**Comment by hbugdoll** [comment]
> [@hansemannn](https://github.com/hansemannn) just saw your commit: [4953156](https://github.com/tidev/titanium-sdk/commit/49531568b4226eed0e5eb84893f7a0f4bfa4da87) Is that fixing this issue here? If so it can be closed

Saw it too. Please leave the issue open.
In my opinion the commit "only" ensures that the `dSYMs` are present in the 3 xcarchives, not in the xcframework.

I've experimented with `-debug-symbols`, but didn't have a ready app update for the App Store to test it with.

**Comment by hansemannn** [comment]
Yes, it's fixed for the main target 👍 TitaniumKit changes are something else

**Comment by hbugdoll** [comment]
> but didn't have a ready app update for the App Store to test it with.

Ah, testing is easy with *Validate App* in Xcode Organizer.

Result of https://github.com/tidev/titanium-sdk/commit/49531568b4226eed0e5eb84893f7a0f4bfa4da87 is as assumed:
<img width="646" height="96" alt="Image" src="https://github.com/user-attachments/assets/6210478c-898a-46f9-a4d4-c223e1dbc9ca" />

I'll provide a PR.
